ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -570,14 +618,7 @@ data ProtocolTransitionParamsShelleyBased era = ProtocolTransitionParamsShelleyB
 -- for mainnet (check against @'SL.gNetworkId' 'shelleyBasedGenesis'@).
 protocolInfoCardano ::
      forall c m. (IOLike m, CardanoHardForkConstraints c)
-  => ProtocolParamsByron
-  -> ProtocolParamsShelleyBased (ShelleyEra c)
-  -> ProtocolParamsShelley c
-  -> ProtocolParamsAllegra c
-  -> ProtocolParamsMary    c
-  -> ProtocolParamsAlonzo  c
-  -> ProtocolParamsBabbage c
-  -> ProtocolParamsConway  c
+  => CardanoProtocolParams c
   -> ProtocolTransitionParamsShelleyBased (ShelleyEra c)
   -> ProtocolTransitionParamsShelleyBased (AllegraEra c)
   -> ProtocolTransitionParamsShelleyBased (MaryEra c)
```

Comment:
INFO: protocol transition parameters should probably also be put in `ProtocolParamsCardano`, in the form of some `InPairs ProtocolTransitionParams (CardanoEras c)` field. However, this requires a new type/data family, so it can probably be done in a follow-up PR. If reviewers agree, I'll open a separate issue for this.

Reply:
Makes sense!

Reply:
Hmm.

```
-- | Parameters needed to transition to a Shelley era.
data ProtocolTransitionParamsShelleyBased era = ProtocolTransitionParamsShelleyBased {
      transitionTranslationContext :: Core.TranslationContext era
    , transitionTrigger            :: TriggerHardFork
    }
```

Nothing about that seems particularly Shelley-specific to me.

Unless `TranslationContext` has some Shelley-specific aspect I'm not remembering.

Or unless we want to explicitly bake-in the fact that we've only ever considered the HFC in the context of Shelley eras.

(Also see my reply to your comment on versions.)

Reply:
I also don't think it looks Shelley-specific, but I also don't know the details about `TranslationContext`. Some experimentation with `data family` again:

```haskell
-- | Parameters needed to transition from one era to the next.
type ProtocolTransitionParams :: Type -> Type -> Type
data family ProtocolTransitionParams x y

-- | Parameters needed to transition from Byron to Shelley.
data instance ProtocolTransitionParams ByronBlock (ShelleyBlock proto era) =
    ProtocolTransitionParamsByronToShelley {
        transitionByronToShelleyTranslationContext :: Core.TranslationContext era
      , transitionByronToShelleyTrigger            :: TriggerHardFork
      }

-- | Parameters needed to transition from any Shelley Era to any Shelley
data instance ProtocolTransitionParams (ShelleyBlock proto era) (ShelleyBlock proto' era') =
    ProtocolTransitionParamsIntraShelley {
        transitionIntraShelleyTranslationContext :: Core.TranslationContext era'
      , transitionIntraShelleyTrigger            :: TriggerHardFork
      }

-- | Parameters needed to run Cardano.
--
-- TODO(jdral): could we get rid of 'shelleyBasedProtocolParams' somehow by
-- including it in the 'ProtocolParams' of one or multiple eras?
data instance ProtocolParams (CardanoBlock c) = ProtocolParamsCardano {
    cardanoProtocolParamsPerEra     :: PerEraProtocolParams (CardanoEras c)
  , shelleyBasedProtocolParams      :: ProtocolParamsShelleyBased (ShelleyEra c)
  , cardanoProtocolTransitionParams :: InPairs ProtocolTransitionParams (CardanoEras c)
  }
```

The nice thing about this is that we can again re-use some `sop-core` machinery: `InPairs`. But maybe we don't need the `data family`, and we can make it a single datatype with the first parameter being a phantom parameter.

Also something I was thinking of: would we want a (phantom) type parameter on `TriggerHardFork`?

Reply:
^ the code is here: https://github.com/input-output-hk/ouroboros-consensus/commit/98ed9447fc35936a18d389f53d8d62f4aa196e1b